karapace/protobuf/message_element.py

```diff
@@ -129,10 +133,21 @@ def compare(self, other: TypeElement, result: CompareResult, types: CompareTypes
                     result.add_modification(Modification.FIELD_ADD)
                 elif other_tags.get(tag) is None:
                     result.add_modification(Modification.FIELD_DROP)
+                elif (
+                    # Check if field tag has been altered
+                    (field_name := self_tags[tag].name)
+                    and (self_field := self_names.get(field_name))
+                    and (other_field := other_names.get(field_name))
+                    and (self_field.tag != other_field.tag)
+                ):
+                    result.pop_path()
+                    result.push_path(field_name)
+                    result.add_modification(Modification.FIELD_TAG_ALTER)
```

Comment:
don't you need to restore the path after pushing the alteration? What happens in neasted messages? Aren't we prepending always that field name to it?
I think we should restore the path after adding the `Modification`

Reply:
I don't think so. In line 149 there is another pop_path called, so immediately after this it will pop and push the next field on the path